packages/fuels-programs/src/calls/contract_call_utils.rs

```diff
@@ -222,87 +210,17 @@ fn generate_custom_outputs(calls: &[ContractCall]) -> Vec<Output> {
         .collect::<Vec<_>>()
 }
 
-fn extract_unique_asset_ids(spendable_coins: &[Resource]) -> HashSet<AssetId> {
-    spendable_coins
-        .iter()
-        .map(|resource| match resource {
-            Resource::Coin(coin) => coin.asset_id,
-            Resource::Message(_) => BASE_ASSET_ID,
-        })
-        .collect()
-}
-
 fn extract_variable_outputs(calls: &[ContractCall]) -> Vec<Output> {
     calls
```

Comment:
```rust
    calls
        .iter()
        .flat_map(|call| &call.variable_outputs)
        .cloned()
        .collect()
```
Perhaps this is better because the original would clone whole vectors and then discard them moving the elements into the final collected vector.

This way you're iterating through each element and cloning the element inside placing it in its final resting place in the collected vector.